### PR TITLE
Add natspec for l2deposit

### PR DESCRIPTION
**Pull request by 108356037**
Add natspec to L2Deposit.sol and IL2Deposit.sol

### CODE REVIEW

File: contracts/interfaces/IL2Deposit.sol [hunk 1]
```diff
@@ -6,6 +6,15 @@ import "./IStrategyBase.sol";
 import "../utils/L2DepositLibEIP712.sol";
 
 interface IL2Deposit is IStrategyBase {
+    /// @notice This event is emmitted when deposit action to layer2 is done
```
Comment (pilagod) [inline contracts/interfaces/IL2Deposit.sol:9]: Maybe stick to the sentence style in LimitOrder like `Emitted when ...`

Comment (pilagod) [reply contracts/interfaces/IL2Deposit.sol:9]: Maybe could reword it like `Emitted when deposit tokens to L2 successfully`

File: contracts/interfaces/IL2Deposit.sol [hunk 2]
```diff
@@ -6,6 +6,15 @@ import "./IStrategyBase.sol";
...
+    /// @param l2Identifier The identifier of which layer2 chain the deposit is send to
```
Comment (pilagod) [inline contracts/interfaces/IL2Deposit.sol:10]: We usually call Layer2 as L2 for brevity. Same for other places

File: contracts/interfaces/IL2Deposit.sol [hunk 3]
```diff
@@ -6,6 +6,15 @@ import "./IStrategyBase.sol";
...
+    /// @param l1TokenAddr The token contract address on layer1
+    /// @param l2TokenAddr The token contract address on layer2
+    /// @param sender The sender's address on layer1
+    /// @param recipient The recipient's address on layer2
+    /// @param amount The amount of token to be sent
+    /// @param data The message data of L2Deposit following EIP-712
```
Comment (pilagod) [inline contracts/interfaces/IL2Deposit.sol:16]: Data is not in EIP712 format, it is just values packed together based on different L2 bridge's needs.

Take this line for example:
https://github.com/consenlabs/tokenlon-contracts/blob/bec978d16b3aed4f720a5debf187b5a59a472bbd/contracts/L2Deposit.sol#L103-L105

File: contracts/interfaces/IL2Deposit.sol [hunk 4]
```diff
@@ -22,5 +31,7 @@ interface IL2Deposit is IStrategyBase {
         bytes depositSig;
     }
 
+    /// @notice Deposit user's fund into layer2 bridge
+    /// @param _params The deposit data following EIP-712 plus the user's signature of it
```
Comment (pilagod) [inline contracts/interfaces/IL2Deposit.sol:35]: `params` is a plain solidity struct. Maybe could take LimitOrder as example to refine the description.

https://github.com/consenlabs/tokenlon-contracts/blob/ece0b1215ea23b095a27e19155f2732d830a2246/contracts/interfaces/ILimitOrder.sol#L122-L133

File: contracts/interfaces/IL2Deposit.sol [hunk 5]
```diff
@@ -22,5 +31,7 @@ interface IL2Deposit is IStrategyBase {
         bytes depositSig;
     }
 
+    /// @notice Deposit user's fund into L2 bridge
+    /// @param _params The deposit data that sends money to L2
```
Comment (pilagod) [inline contracts/interfaces/IL2Deposit.sol:35]: nit: It might be more suitable to use `token` rather than `money` 😆

**Review by pilagod — APPROVED** [review]

File: contracts/interfaces/IL2Deposit.sol [hunk 6]
```diff
@@ -6,6 +6,15 @@ import "./IStrategyBase.sol";
 import "../utils/L2DepositLibEIP712.sol";
 
 interface IL2Deposit is IStrategyBase {
+    /// @notice Emmitted when deposit tokens to L2 successfully
```
Comment (NIC619) [inline contracts/interfaces/IL2Deposit.sol:9]:
```suggestion
    /// @notice Emitted when deposit tokens to L2 successfully
```

File: contracts/interfaces/IL2Deposit.sol [hunk 7]
```diff
@@ -6,6 +6,15 @@ import "./IStrategyBase.sol";
...
+    /// @param l2Identifier The identifier of which L2 chain the deposit is send to
```
Comment (NIC619) [inline contracts/interfaces/IL2Deposit.sol:10]:
```suggestion
    /// @param l2Identifier The identifier of which L2 chain the deposit is sent to
```

**Review by pilagod — APPROVED** [review]

**Review by NIC619 — APPROVED** [review]

**Pull request merged**